### PR TITLE
CLI for TWAP between historic time intervals

### PR DESCRIPTION
**Pull request by hieuvubk**
Closes: #5130 

## What is the purpose of the change

Currently, we only keep the twap record of the last 2 days, which causes difficulties if we try to query twap outside this time period.

This pr introduces a new cli called `extra-arithmetic` that can query twap of any time range without limitation. Basically, It will separate the input time range into 2-day intervals, take the twap of each interval and then calculate the avarage value.

## Testing and Verifying

Tested with mainnet data

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes?
  - [ ] Changelog entry added to `Unreleased` section of `CHANGELOG.md`?

Where is the change documented? 
  - [ ] Specification (`x/{module}/README.md`)
  - [ ] Osmosis documentation site
  - [ ] Code comments?
  - [ ] N/A

### CODE REVIEW

**Comment by hieuvubk** [comment]
@p0mvn I think the `extra-arithmetic` should work rn, could u check it. I tried query with start time at `5 October 2023 09:49:09` (records must be prun) & interval 50h and here the result:

```
osmosisd query twap extra-arithmetic 1 uosmo 1696499349 50h --node https://rpc.osmosis.zone:443
arithmetic_twap: "0.042952026865644822"
```

**Comment by czarcas7ic** [comment]
@hieuvubk can you please fill out the PR description, as well as add tags necessary to allow CI to pass?

I generally quickly look through the PR list, and if the description isn't set and/or the CI isn't passing I tend to just pass over it.

**Comment by hieuvubk** [comment]
> @hieuvubk can you please fill out the PR description, as well as add tags necessary to allow CI to pass?
> 
> I generally quickly look through the PR list, and if the description isn't set and/or the CI isn't passing I tend to just pass over it.

Sr for late reply, I forgot to add to changelog. Also update description & changelog both. Ty for reminder

File: x/twap/client/cli/query.go [hunk 1]
```diff
@@ -83,6 +85,149 @@ Example:
 	return cmd
 }
 
+// GetQueryExtraArithmeticCommand returns an arithmetic twap query command with no [start, end] range limit.
+func GetQueryExtraArithmeticCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "extra-arithmetic [poolid] [base denom] [start time] [end time]",
```
Comment (mattverse) [inline x/twap/client/cli/query.go:91]:
```suggestion
		Use:     "extra-arithmetic",
```

File: x/twap/client/cli/query.go [hunk 2]
```diff
@@ -83,6 +85,149 @@ Example:
...
+	ctx := client.GetClientContextFromCmd(cmd)
+	ctx = ctx.WithHeight(height)
+	return client.ReadPersistentCommandFlags(ctx, cmd.Flags())
+}
+
+// Try to find the block containing the given timestamp
+func findBlockByTime(clientCtx client.Context, time time.Time, currentHeight int64) (int64, error) {
+	// Currently we hardcode this value
+	// TODO: Fetch block time from somewhere
+	blockTime := 5.84
```
Comment (mattverse) [inline x/twap/client/cli/query.go:190]: Hrmmm, I'm not sure if the result that came from this hard coded value would have any significant meaning 🤔

File: x/twap/client/cli/query.go (hunk 1 above)
Comment (hieuvubk) [reply x/twap/client/cli/query.go:91]: Why we need to remove args here?

File: x/twap/client/cli/query.go (hunk 2 above)
Comment (hieuvubk) [reply x/twap/client/cli/query.go:190]: I take this value from mintscan, its a way for me to find past blocks based on the current time

File: x/twap/client/cli/query.go (hunk 1 above)
Comment (mattverse) [reply x/twap/client/cli/query.go:91]: it gets auto-generated

File: x/twap/client/cli/query.go (hunk 2 above)
Comment (p0mvn) [reply x/twap/client/cli/query.go:190]: Can we programmatically fetch it from somewhere? E.g. take 10 samples in the past 1000 heights and average them out

**Pull request closed without merging**